app/models/concerns/taggable.rb

```diff
@@ -6,32 +6,33 @@ module Taggable
     has_many :tags, through: :taggings
   end
 
-  def tag_as(name)
+  def tag_with(tags = nil, name: nil)
```

Comment:
i think this might be a little cleaner (it's untested tho)
```ruby
def tag_with(tags = [], name: nil)
  tags = tags.to_a
  raise ArgumentError unless tags.any? || name

  tags << Tag.find_by(name:) if name
  tags.each do |tag|
    taggings.create_or_find_by! tag: tag
  end
end

def tag_with!(tags = [], name: nil)
  tags = tags.to_a
  tags << Tag.find_or_create_by! name: if name

  tag_with tags
end
```

Reply:
This allows for use like `#tag_with Tag.first, name: "(another tag)"` which would lead to more confusion. Maybe we just allow for a `tags_or_name`.